net: openthread: Fix exiting diag when transmission is ongoing

MarekPorwisz opened the pull request:
Crash was observed if "diag stop" was invoked during the transmission. The issue was caused by an update to otPlatRadioSleep that made it compilant with the documentation and not allowing direct Transmeet to sleep transition.
The transmission result was forwarded to OpenThread making it crash due to an invalid state.

rlubos commented on modules/openthread/platform/diag.c:
```diff
@@ -92,18 +92,25 @@ otError otPlatDiagProcess(otInstance *aInstance, uint8_t aArgsLength, char *aArg
 }
 
-void otPlatDiagModeSet(bool aMode)
+static void log_error_returned(otError aError, const char * aName)
```
Please use snake case for Zephyr code (i.e. functions not implementing OT API), so `error`, `name`

MarekPorwisz committed:
net: openthread: Fix exiting diag when transmission is ongoing

Crash was observed if "diag stop" was invoked during the transmission.
The issue was caused by an update to otPlatRadioSleep that made it
compilant with the documentation and not allowing direct Transmit to
Sleep transition.
The transmission result was forwarded to OpenThread making it crash due
to an invalid state.

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>
